### PR TITLE
[android] update built-in AOT profiles

### PR DESCRIPTION
**Pull request by jonathanpeppers**
Since ceaff6b3 added Shell to the template, startup times regressed:

    03-09 13:35:55.695  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +825ms
    03-09 13:35:56.871  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +814ms
    03-09 13:35:58.050  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +803ms
    03-09 13:35:59.255  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +818ms
    03-09 13:36:00.508  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +832ms
    03-09 13:36:01.711  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +843ms
    03-09 13:36:02.915  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +824ms
    03-09 13:36:04.138  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +835ms
    03-09 13:36:05.329  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +807ms
    03-09 13:36:06.570  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +855ms
    Average(ms): 825.6
    Std Err(ms): 5.10816340119747
    Std Dev(ms): 16.1534310080965

Part of the problem here is the startup path changed dramatically --
and so many methods are being JIT'd at startup that were not even
called before.

Additionally, I wanted to add a few things to the profile that make it
generally better for "real" apps:

1. Since c40c6e77 removed all culture-aware string comparisons, I
   added some in the profile to keep more BCL culture-related types.

2. Use `{OnPlatform}`, as it shows up in the .NET podcast JIT times.

3. Use `Task.Delay()` and `Task.CompletedTask` as these also show up in
   the .NET podcast JIT times.

4. Use some string interpolation, like `$""`.

With an updated profile, a `Release` build of a `dotnet new maui`
project on a Pixel 5 now gets:

    03-09 13:58:44.428  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +585ms
    03-09 13:58:45.702  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +589ms
    03-09 13:58:46.916  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +598ms
    03-09 13:58:48.134  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +616ms
    03-09 13:58:49.317  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +600ms
    03-09 13:58:50.554  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +591ms
    03-09 13:58:51.781  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +590ms
    03-09 13:58:52.989  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +587ms
    03-09 13:58:54.178  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +589ms
    03-09 13:58:55.419  1802  1962 I ActivityTaskManager: Displayed com.companyname.foo/crc64808a40cc7e533249.MainActivity: +623ms
    Average(ms): 596.8
    Std Err(ms): 4.08737894825849
    Std Dev(ms): 12.9254271367203

The .NET Podcast app is getting:

    03-09 14:06:39.197  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s139ms
    03-09 14:06:41.400  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s150ms
    03-09 14:06:43.590  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s151ms
    03-09 14:06:45.783  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s128ms
    03-09 14:06:48.009  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s143ms
    03-09 14:06:50.173  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s123ms
    03-09 14:06:52.383  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s167ms
    03-09 14:06:54.544  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s138ms
    03-09 14:06:56.795  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s176ms
    03-09 14:06:58.938  1802  1962 I ActivityTaskManager: Displayed com.Microsoft.NetConf2021.Maui/crc64b93ba739210fe662.MainActivity: +1s152ms
    Average(ms): 1146.7
    Std Err(ms): 5.15115952426679
    Std Dev(ms): 16.2893966875524

### CODE REVIEW

File: src/ProfiledAot/src/CommonMethods.cs [hunk 1]
```diff
@@ -1,19 +1,27 @@
 // This class represents common code paths we always want to AOT
 
+// string interpolation
+// Culture-aware string comparisons
 // System.Threading.Tasks.Task
 // System.Net.Http.HttpClient
 
+#pragma warning disable CA1307
```
Comment (eerhardt) [inline src/ProfiledAot/src/CommonMethods.cs:8]: Can we add a comment here explaining why?

File: src/ProfiledAot/src/CommonMethods.cs [hunk 2]
```diff
@@ -1,19 +1,27 @@
 // This class represents common code paths we always want to AOT
 
+// string interpolation
+// Culture-aware string comparisons
 // System.Threading.Tasks.Task
 // System.Net.Http.HttpClient
```
Comment (eerhardt) [inline src/ProfiledAot/src/CommonMethods.cs:6]: Were you going to also add MS.Extensions.Http?  or is that in a future change?

File: src/ProfiledAot/src/maui/App.xaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            <Color x:Key="DarkColor">Black</Color>
+
+            <Style TargetType="Page" ApplyToDerivedTypes="True">
+                <Setter Property="BackgroundColor"
+                        Value="{AppThemeBinding Light={StaticResource LightColor}, Dark={StaticResource DarkColor}}" />
+            </Style>
+
+            <Style TargetType="Label">
+                <Setter Property="TextColor" Value="{StaticResource PrimaryColor}" />
+                <!-- This line is the only change from the template, to use {OnPlatform} -->
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/ProfiledAot/src/maui/App.xaml:23]: It looks like the two templates have very different `App.xaml` files right now.

[the remark this answers is not in the document]
Comment (eerhardt) [reply src/ProfiledAot/src/maui/App.xaml:23]: But can we just make the change in one of them? So we don't have to maintain copies of both?

File: src/ProfiledAot/src/CommonMethods.cs (hunk 2 above)
Comment (jonathanpeppers) [reply src/ProfiledAot/src/CommonMethods.cs:6]: After testing some more, I don't find any `AOT NOT FOUND` messages for this in the Podcast app.

Every line with "http" is:
````
03-09 14:58:41.368 11672 11672 D Mono    : AOT NOT FOUND: (wrapper managed-to-managed) Microsoft.Extensions.Http.IHttpMessageHandlerBuilderFilter[]:System.Collections.Generic.ICollection`1.get_Count ().
03-09 14:58:41.516 11672 11672 D Mono    : AOT NOT FOUND: Microsoft.Extensions.Logging.Logger:BeginScope<Microsoft.Extensions.Logging.LoggerMessage/LogValues`2<System.Net.Http.HttpMethod, System.Uri>> (Microsoft.Extensions.Logging.LoggerMessageLogValues`2<System.Net.Http.HttpMethod, System.Uri>).
03-09 14:58:41.530 11672 11709 D Mono    : AOT NOT FOUND: System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1/AsyncStateMachineBox`1<Java.Net.HttpURLConnection, Xamarin.Android.Net.AndroidMessageHandler/<SetupRequestInternal>d__146>:ExecuteFromThreadPool (System.Threading.Thread).
03-09 14:58:41.531 11672 11672 D Mono    : AOT NOT FOUND: System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Collections.Generic.IEnumerable`1<Microsoft.NetConf2021.Maui.Models.Responses.ShowResponse>>:AwaitUnsafeOnCompleted<System.Runtime.CompilerServices.TaskAwaiter`1<System.Net.Http.HttpResponseMessage>, Microsoft.NetConf2021.Maui.Services.ShowsService/<TryGetAsync>d__9`1<System.Collections.Generic.IEnumerable`1<Microsoft.NetConf2021.Maui.Models.Responses.ShowResponse>>> (System.Runtime.CompilerServices.TaskAwaiter`1<System.Net.Http.HttpResponseMessage>&,Microsoft.NetConf2021.Maui.Services.ShowsService/<TryGetAsync>d__9`1<System.Collections.Generic.IEnumerable`1<Microsoft.NetConf2021.Maui.Models.Responses.ShowResponse>>&).
03-09 14:58:41.533 11672 11672 D Mono    : AOT NOT FOUND: System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Collections.Generic.IEnumerable`1<Microsoft.NetConf2021.Maui.Models.Responses.ShowResponse>>:AwaitUnsafeOnCompleted<System.Runtime.CompilerServices.TaskAwaiter`1<System.Net.Http.HttpResponseMessage>> (System.Runtime.CompilerServices.TaskAwaiter`1<System.Net.Http.HttpResponseMessage>&,System.Runtime.CompilerServices.IAsyncStateMachineBox).
````

File: src/ProfiledAot/src/maui/App.xaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/ProfiledAot/src/maui/App.xaml:23]: Ah, yes. I can probably just do it in one of the profiles.

**Review by eerhardt — APPROVED** [review]
:shipit:

**Pull request merged**